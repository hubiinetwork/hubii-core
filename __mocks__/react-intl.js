const Intl = jest.genMockFromModule('react-intl');

Intl.intl = {
  formatMessage: ({id}) => id
};

Intl.injectIntl = (Node) => Node

Intl.defineMessages = (obj) => obj

module.exports = Intl;