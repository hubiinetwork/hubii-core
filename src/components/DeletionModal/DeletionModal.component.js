import React from 'react';
import PropTypes from 'prop-types';
import {
  Text,
  TextPrimary,
  Wrapper,
  StyledButton,
  ParentDiv,
  SecondaryHeader,
} from './DeletionModal.style';


/**
 * DeletionModal
 */
const DeletionModal = (props) => (
  <Wrapper>
    <Text> Are you sure you want to delete this {props.type}?</Text>
    <TextPrimary>
      <div>
        <SecondaryHeader>Name</SecondaryHeader> {props.name}
      </div>
      <div>
        <SecondaryHeader>Address</SecondaryHeader>{props.address}
      </div>
    </TextPrimary>
    <ParentDiv>
      <StyledButton type="danger" onClick={props.onDelete} id="delete">
        Delete
      </StyledButton>
      <StyledButton type="default" onClick={props.onCancel} id="cancel">
        Cancel
      </StyledButton>
    </ParentDiv>
  </Wrapper>
  );
DeletionModal.propTypes = {
  /**
   * Function to perform action when cancel button is clicked
   */
  onCancel: PropTypes.func,
  /**
   * Function to perform action when delete button is clicked
   */
  onDelete: PropTypes.func,
  /**
   * Name
   */
  name: PropTypes.string,
  /**
   * Address
   */
  address: PropTypes.string,
  /**
   * Type of Deletion (i.e "wallet", "contact")
   */
  type: PropTypes.string.isRequired,
};
export default DeletionModal;
