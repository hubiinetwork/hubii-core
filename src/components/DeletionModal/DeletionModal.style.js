import styled from 'styled-components';
import Button from 'components/ui/Button';

export const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
`;
export const Text = styled.div`
  color: ${({ theme }) => theme.palette.info};
  font-size: large;
  margin-bottom: 1.5rem;
`;

export const SecondaryHeader = styled.div`
  color: ${({ theme }) => theme.palette.info};
  font-size: large;
  margin: 0.5rem 0 0.5rem;
`;
export const TextPrimary = styled.div`
  color: ${({ theme }) => theme.palette.light};
  text-align: center;
`;
export const StyledButton = styled(Button)`
  min-width: 10.71rem;
  padding: 0.5rem 1rem;
  margin: 1rem;
`;
export const ParentDiv = styled.div`
  margin-top: 1.43rem;
  display: flex;
  justify-content: center;
`;
