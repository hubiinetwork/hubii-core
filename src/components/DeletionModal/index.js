export { default } from './DeletionModal.component';
