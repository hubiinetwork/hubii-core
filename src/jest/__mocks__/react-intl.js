const Intl = jest.genMockFromModule('react-intl');

Intl.intl = {
  formatMessage: ({ id }, params) => params ? `${id} ${JSON.stringify(params)}` : id,
};

Intl.injectIntl = (Node) => Node;

Intl.defineMessages = (obj) => obj;

module.exports = Intl;
