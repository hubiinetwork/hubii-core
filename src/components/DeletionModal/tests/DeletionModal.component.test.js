import React from 'react';
import { shallow } from 'enzyme';
import DeletionModal from 'components/DeletionModal';
import { intl } from '../../../../__mocks__/react-intl';

describe('<ContactDeletionModal/>', () => {
  const contact = {
    name: 'mike',
    address: '0x3123123',
  };
  const type = 'wallet';
  const onCancel = jest.fn();
  const onDelete = jest.fn();

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <DeletionModal
        name={contact.name}
        address={contact.address}
        onCancel={onCancel}
        onDelete={onDelete}
        type={type}
        intl={intl}
      />
    );
  });

  it('should render <DeletionModal/> correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('when the delete button is clicked the onDelete function is run', () => {
    const button = wrapper.find('#delete');
    button.simulate('click');
    expect(onDelete).toHaveBeenCalledTimes(1);
  });

  it('when the cancel button is clicked the onCancel function is run', () => {
    const button = wrapper.find('#cancel');
    button.simulate('click');
    expect(onCancel).toHaveBeenCalledTimes(1);
  });
});
