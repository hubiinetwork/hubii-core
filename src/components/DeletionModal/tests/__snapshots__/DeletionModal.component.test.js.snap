// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContactDeletionModal/> should render <DeletionModal/> correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DeletionModal
    address="0x3123123"
    intl={
        Object {
            "formatMessage": [Function],
          }
    }
    name="mike"
    onCancel={[Function]}
    onDelete={[Function]}
    type="wallet"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <DeletionModal.style__TopHeading>
           
          delete_confirm {"type":"wallet"}
</DeletionModal.style__TopHeading>,
        <DeletionModal.style__TextPrimary>
          <div
                    style={
                              Object {
                                        "marginBottom": "0.5rem",
                                      }
                    }
          >
                    <DeletionModal.style__SecondaryHeader
                              large={true}
                    >
                              name
                    </DeletionModal.style__SecondaryHeader>
                    <br />
                    <Text>
                              mike
                    </Text>
          </div>
          <div>
                    <DeletionModal.style__SecondaryHeader
                              large={true}
                    >
                              address
                    </DeletionModal.style__SecondaryHeader>
                    <br />
                    <Text>
                              0x3123123
                    </Text>
          </div>
</DeletionModal.style__TextPrimary>,
        <DeletionModal.style__ParentDiv>
          <DeletionModal.style__StyledButton
                    id="delete"
                    onClick={[Function]}
                    type="danger"
          >
                    delete
          </DeletionModal.style__StyledButton>
          <DeletionModal.style__StyledButton
                    id="cancel"
                    onClick={[Function]}
                    type="default"
          >
                    cancel
          </DeletionModal.style__StyledButton>
</DeletionModal.style__ParentDiv>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            " ",
            "delete_confirm {\\"type\\":\\"wallet\\"}",
          ],
        },
        "ref": null,
        "rendered": Array [
          " ",
          "delete_confirm {\\"type\\":\\"wallet\\"}",
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <div
              style={
                            Object {
                                          "marginBottom": "0.5rem",
                                        }
              }
>
              <DeletionModal.style__SecondaryHeader
                            large={true}
              >
                            name
              </DeletionModal.style__SecondaryHeader>
              <br />
              <Text>
                            mike
              </Text>
</div>,
            <div>
              <DeletionModal.style__SecondaryHeader
                            large={true}
              >
                            address
              </DeletionModal.style__SecondaryHeader>
              <br />
              <Text>
                            0x3123123
              </Text>
</div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <DeletionModal.style__SecondaryHeader
                  large={true}
>
                  name
</DeletionModal.style__SecondaryHeader>,
                <br />,
                <Text>
                  mike
</Text>,
              ],
              "style": Object {
                "marginBottom": "0.5rem",
              },
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "name",
                  "large": true,
                },
                "ref": null,
                "rendered": "name",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "mike",
                },
                "ref": null,
                "rendered": "mike",
                "type": [Function],
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <DeletionModal.style__SecondaryHeader
                  large={true}
>
                  address
</DeletionModal.style__SecondaryHeader>,
                <br />,
                <Text>
                  0x3123123
</Text>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "address",
                  "large": true,
                },
                "ref": null,
                "rendered": "address",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "0x3123123",
                },
                "ref": null,
                "rendered": "0x3123123",
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <DeletionModal.style__StyledButton
              id="delete"
              onClick={[Function]}
              type="danger"
>
              delete
</DeletionModal.style__StyledButton>,
            <DeletionModal.style__StyledButton
              id="cancel"
              onClick={[Function]}
              type="default"
>
              cancel
</DeletionModal.style__StyledButton>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "delete",
              "id": "delete",
              "onClick": [Function],
              "type": "danger",
            },
            "ref": null,
            "rendered": "delete",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "cancel",
              "id": "cancel",
              "onClick": [Function],
              "type": "default",
            },
            "ref": null,
            "rendered": "cancel",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <DeletionModal.style__TopHeading>
             
            delete_confirm {"type":"wallet"}
</DeletionModal.style__TopHeading>,
          <DeletionModal.style__TextPrimary>
            <div
                        style={
                                    Object {
                                                "marginBottom": "0.5rem",
                                              }
                        }
            >
                        <DeletionModal.style__SecondaryHeader
                                    large={true}
                        >
                                    name
                        </DeletionModal.style__SecondaryHeader>
                        <br />
                        <Text>
                                    mike
                        </Text>
            </div>
            <div>
                        <DeletionModal.style__SecondaryHeader
                                    large={true}
                        >
                                    address
                        </DeletionModal.style__SecondaryHeader>
                        <br />
                        <Text>
                                    0x3123123
                        </Text>
            </div>
</DeletionModal.style__TextPrimary>,
          <DeletionModal.style__ParentDiv>
            <DeletionModal.style__StyledButton
                        id="delete"
                        onClick={[Function]}
                        type="danger"
            >
                        delete
            </DeletionModal.style__StyledButton>
            <DeletionModal.style__StyledButton
                        id="cancel"
                        onClick={[Function]}
                        type="default"
            >
                        cancel
            </DeletionModal.style__StyledButton>
</DeletionModal.style__ParentDiv>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              " ",
              "delete_confirm {\\"type\\":\\"wallet\\"}",
            ],
          },
          "ref": null,
          "rendered": Array [
            " ",
            "delete_confirm {\\"type\\":\\"wallet\\"}",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <div
                style={
                                Object {
                                                "marginBottom": "0.5rem",
                                              }
                }
>
                <DeletionModal.style__SecondaryHeader
                                large={true}
                >
                                name
                </DeletionModal.style__SecondaryHeader>
                <br />
                <Text>
                                mike
                </Text>
</div>,
              <div>
                <DeletionModal.style__SecondaryHeader
                                large={true}
                >
                                address
                </DeletionModal.style__SecondaryHeader>
                <br />
                <Text>
                                0x3123123
                </Text>
</div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <DeletionModal.style__SecondaryHeader
                    large={true}
>
                    name
</DeletionModal.style__SecondaryHeader>,
                  <br />,
                  <Text>
                    mike
</Text>,
                ],
                "style": Object {
                  "marginBottom": "0.5rem",
                },
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "name",
                    "large": true,
                  },
                  "ref": null,
                  "rendered": "name",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "br",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "mike",
                  },
                  "ref": null,
                  "rendered": "mike",
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <DeletionModal.style__SecondaryHeader
                    large={true}
>
                    address
</DeletionModal.style__SecondaryHeader>,
                  <br />,
                  <Text>
                    0x3123123
</Text>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "address",
                    "large": true,
                  },
                  "ref": null,
                  "rendered": "address",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "br",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "0x3123123",
                  },
                  "ref": null,
                  "rendered": "0x3123123",
                  "type": [Function],
                },
              ],
              "type": "div",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <DeletionModal.style__StyledButton
                id="delete"
                onClick={[Function]}
                type="danger"
>
                delete
</DeletionModal.style__StyledButton>,
              <DeletionModal.style__StyledButton
                id="cancel"
                onClick={[Function]}
                type="default"
>
                cancel
</DeletionModal.style__StyledButton>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "delete",
                "id": "delete",
                "onClick": [Function],
                "type": "danger",
              },
              "ref": null,
              "rendered": "delete",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "cancel",
                "id": "cancel",
                "onClick": [Function],
                "type": "default",
              },
              "ref": null,
              "rendered": "cancel",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
